storage/s3.go

```diff
@@ -0,0 +1,135 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/Scalingo/go-utils/logger"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/aws/defaults"
+	"github.com/aws/aws-sdk-go-v2/aws/endpoints"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3manager"
+	"github.com/pkg/errors"
+)
+
+const (
+	NotFoundErrCode = "NotFound"
+)
+
+type S3Client interface {
+	GetObjectRequest(input *s3.GetObjectInput) s3.GetObjectRequest
+	HeadObjectRequest(input *s3.HeadObjectInput) s3.HeadObjectRequest
+	DeleteObjectRequest(input *s3.DeleteObjectInput) s3.DeleteObjectRequest
+}
+
+type S3Config struct {
+	AK       string
+	SK       string
+	Region   string
+	Endpoint string
+	Bucket   string
+}
+
+type S3 struct {
+	cfg               S3Config
+	s3client          S3Client
+	s3uploader        *s3manager.Uploader
+	retryWaitDuration time.Duration
+	retryAttempts     int
+}
+
+func NewS3(cfg S3Config, bucket string) *S3 {
+	s3config := s3Config(cfg)
+	return &S3{
+		cfg: cfg, s3client: s3.New(s3config), s3uploader: s3manager.NewUploader(s3config),
+		retryWaitDuration: time.Second, retryAttempts: 3,
```

Comment:
Shouldn't those two fields be configurable ?

Reply:
Actually after thinking I'm wondering if it's not the package use who should be responsible of retrying a Size() command (waiting for eventual consistency right after an upload) What do you think?

Reply:
mmmh if the service is eventually consistent i find it pretty nice that the package offer this option. Otherwise it should be configurable.

Reply:
Hmmm ok :-)